### DISCUSSION TITLE
What will happen when submit multi run jobs from bloop cli?

### DISCUSSION THREAD

**Discussion by davidmlw** [scalacenter/bloop#1291] [General]
A well prepared workspace with multiple projects. If the following command is typed mannually in a terminal, it is passed.

```
bloop compile topProject
bloop run -p TopProject mainA args
bloop run -p TopProject mainB argsc
bloop run -p TopProject mainC args
```

But what will happen if multiple run job submitted simultaneously by a program? Could they be locked by each other? Is bloop designed to be used in this way? Is Bloop an acctually a Job Scheduler too?

```
bloop run -p TopProject mainA args
bloop run -p TopProject mainB argsc
bloop run -p TopProject mainC args
```

**Comment by jvican** [comment]
Good questions. These are answered in our "What is Bloop" page, see https://scalacenter.github.io/bloop/docs/what-is-bloop#the-goal-of-a-build-server. In short, bloop supports different, concurrent clients asking to run an application just fine, and the compilation (if there are things to compile) will only occur once and will be deduplicated for the rest of the clients.
